### ISSUE TITLE
[BUG] Can't install git module - npm ERR! Unsupported URL Type "ssh:": ssh://git...

### ISSUE THREAD

**Issue by frankra** [npm/cli#3537]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have a project that uses git modules:

```
package.json
{
    "name": "my-project",
    "version": "1.0.0",
    "dependencies": {
        "my-internal-module": "git+ssh://git@github.internal.corp:org/my-internal-module.git#d3d8519716feef286c97dc3991a53df5b5e5e704",
        ...
    },
    ...
}
```

I removed both `node_modules` and `package-lock.json` to be able to fix vulnerable dependencies, and I've been trying to install everything again with `npm i` but it is failing with the following error:

```
➜  my-project git:(feat/update-deps) ✗ npm i
npm ERR! code EUNSUPPORTEDPROTOCOL
npm ERR! Unsupported URL Type "ssh:": ssh://git@github.internal.corp:org/my-internal-module.git#d3d8519716feef286c97dc3991a53df5b5e5e704

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/you/.npm/_logs/2021-07-13T09_30_42_819Z-debug.log
```

### Expected Behavior

It should just install the module as it used to.

### Steps To Reproduce

Is not really easy to reproduce as you need an internally/corporate hosted github...
I tried the same with the public github.com and it works. See: https://github.com/frankra/npm-can-install-git-module

1. In a project that has a dependency from a module hosted in an internally hosted github
2. delete both `node_modules` and `package-lock.json`
3. Run `npm i`
4. See error...

### Environment

- OS: macOS Big Sur (11.4)
- Node: 14.15.4
- npm: 7.15.1

**Comment by frankra** [comment]
I was able to solve this issue by changing the dependency url from:

```
"my-internal-module": "git+ssh://git@github.internal.corp:org/my-internal-module.git#d3d8519716feef286c97dc3991a53df5b5e5e704",
```

to

```
 "my-internal-module": "github.internal.corp:org/my-internal-module.git#d3d8519716feef286c97dc3991a53df5b5e5e704",
```

But this is still a bet weird because as per the [documentation](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#git-urls-as-dependencies), both of them should work...

just removed the `git+ssh://git@` part.

**Comment by lmartorella** [comment]
I reproduce the same issue (when the private git requires a different port, like:

```
git+ssh://git@git.mycompany.com:1234/group/repo.git
```

Result:
```
npm ERR! code EUNSUPPORTEDPROTOCOL
npm ERR! Unsupported URL Type "ssh:": ssh://git@git.mycompany.com:1234/group/repo.git
```

npm 7.24.2. Latest npm 6 works fine.

**Comment by matthi4s** [comment]
That should be fixed in npm [8.1.0](https://github.com/npm/cli/releases/tag/v8.1.0), see https://github.com/npm/pacote/pull/86.

**Comment by nlf** [comment]
as mentioned, this should be fixed as of npm 8.1.0. if you experience this problem again, please open a new issue and provide as much detail as possible.
